pkg/frontend/querymiddleware/model_extra.go

```diff
@@ -332,19 +332,13 @@ func (r *PrometheusInstantQueryRequest) GetStep() int64 {
 // GetMinT returns the minimum timestamp in milliseconds of data to be queried,
 // as determined from the start timestamp and any range vector or offset in the query.
 func (r *PrometheusInstantQueryRequest) GetMinT() int64 {
-	minT, _ := decodeQueryMinMaxTime(
-		r.queryExpr, r.GetStart(), r.GetEnd(), r.GetStep(), r.lookbackDelta,
-	)
-	return minT
+	return r.minT
```

Comment:
Check all places where `queryExpr` is update in `PrometheusInstantQueryRequest` functions. They all have the condition `if newRequest.queryExpr != nil {` (I still don't understand in which case it could be nil). If it's nil, then `mint` and `maxt` are not updated, but they should be reset to `time` like we do in `NewPrometheusInstantQueryRequest`.

Unrelated from your PR: I think the logic in the `PrometheusInstantQueryRequest.With...()` functions is a bit fragile. They all copy-paste a bunch of code and they're not even consistent with the requirement of doing a deep copy. In a separate PR I would try to further cleanup this implementation.

Reply:
now in the protection of unit tests, factored out the minT, maxT update: https://github.com/grafana/mimir/pull/8443/commits/e2bdbabd9fb8b994bf094f3bb7a37cacb411a188